tests/api_connexion/endpoints/test_xcom_endpoint.py

```diff
@@ -15,62 +15,254 @@
         response = self.client.get(
-            "/dags/TEST_DAG_ID}/taskInstances/TEST_TASK_ID/2005-04-02T21:37:42Z/xcomEntries/"
+            f"/api/v1/dags/{dag_id}/dagRuns/{dag_run_id}/taskInstances/{task_id}/xcomEntries"
+        )
+        self.assertEqual(200, response.status_code)
+        self.assertEqual(
+            response.json,
+            {
+                'xcom_entries': [
+                    {
+                        'dag_id': dag_id,
+                        'execution_date': execution_date,
+                        'key': 'test-xcom-key-1',
+                        'task_id': task_id,
+                        'timestamp': execution_date
+                    },
+                    {
+                        'dag_id': dag_id,
+                        'execution_date': execution_date,
+                        'key': 'test-xcom-key-2',
+                        'task_id': task_id,
+                        'timestamp': execution_date
+                    }
+                ],
+                'total_entries': 2,
+            }
+        )
+
+
+class TestPaginationGetXComEntries(TestXComEndpoint):
+
+    def setUp(self):
+        super().setUp()
+        self.dag_id = 'test-dag-id'
+        self.task_id = 'test-task-id'
+        self.execution_date = '2005-04-02T21:37:42+00:00'
+        self.execution_date_parsed = parse_execution_date(self.execution_date)
+        self.dag_run_id = DR.generate_run_id(DagRunType.MANUAL, self.execution_date_parsed)
+        self.expected_xcom_ids = sorted([
+            'TEST_XCOM_KEY1',
+            'TEST_XCOM_KEY2',
+            'TEST_XCOM_KEY3',
+            'TEST_XCOM_KEY4',
+            'TEST_XCOM_KEY5',
+            'TEST_XCOM_KEY6',
+            'TEST_XCOM_KEY7',
+            'TEST_XCOM_KEY8',
+            'TEST_XCOM_KEY9',
+            'TEST_XCOM_KEY10',
+        ])
+
+    @parameterized.expand(
+        [
+            (
+                "limit=1"
+            ),
+            (
+                "limit=2"
+            ),
+            (
+                "offset=5"
+            ),
+            (
+                "offset=0"
+            ),
+            (
+                "limit=1&offset=5"
+            ),
+            (
+                "limit=1&offset=1"
+            ),
+            (
+                "limit=2&offset=2"
+            ),
+        ]
+    )
+    @provide_session
+    def test_handle_limit_offset(self, query_params, session):
+        url = "/api/v1/dags/{dag_id}/dagRuns/{dag_run_id}/taskInstances/{task_id}/xcomEntries?{query_params}"
+        url = url.format(dag_id=self.dag_id,
+                         dag_run_id=self.dag_run_id,
+                         task_id=self.task_id,
+                         query_params=query_params)
+        parsed = dict(parse.parse_qsl(parse.urlsplit(url).query))
+        limit = int(parsed.get('limit', 100))
+        offset = int(parsed.get('offset', 0))
+        expected_xcom_ids = self.expected_xcom_ids[0 + offset: offset + limit]
```

Comment:
This should be a constant value. Whenever possible, the tests should only create variables, call other functions, do assertions. They should not calculate anything because you could have made a mistake in two places. 

https://testing.googleblog.com/2019/12/testing-on-toilet-tests-too-dry-make.html